### ISSUE TITLE
Migrations for abi_migration_branches dependent on main branch requirements

### ISSUE THREAD

**Issue by mbargull** [regro/cf-scripts#1979]
It looks as if migration PRs are only issued to `abi_migration_branches` if a feedstock's `main` branch is picked up for migration.

Reported at https://github.com/conda-forge/r-base-feedstock/pull/279#issuecomment-1861409223 .

 I'm not familiar with the graph construction, but glancing at https://github.com/regro/cf-graph-countyfair/blob/d57ac175c0d94685be71c830a7ff953898323363/node_attrs/8/b/9/8/f/r-base.json , I'd assume we'd need `requirements` information for the `abi_migration_branches` in there.
Alternatively (and possibly cleaner) we'd need multiple nodes for the different branches, e.g., `r-base@main.json`, `r-base@4.2.x.json`, `r-base@4.1.x.json` for the `r-base` case.
The `.json`s already have a `branch` key in there, which seem to offer itself for such a structure.
(But the above is really just a daring claim with not really knowing how things work...)

cc @mfansler , @jdblischak

**Comment by mfansler** [comment]
Thanks for following up on this @mbargull - wasn't exactly sure where to report.

**Comment by beckermr** [comment]
Yes this is the currently implemented behavior. It won't be easy to fix either.

**Comment by mbargull** [comment]
Not want I wanted to hear -- but kind of what I expected :D.
Thanks for confirming, appreciated!
